src/inet/TCPEndPoint.h

```diff
@@ -625,5 +628,12 @@ struct EndPointProperties<TCPEndPoint>
     static constexpr int kSystemStatsKey  = System::Stats::kInetLayer_NumTCPEps;
 };
 
+template <>
+class EndPointDeletor<TCPEndPoint>
+{
+public:
+    static void Release(TCPEndPoint * obj) { obj->Free(); }
```

Comment:
This basically is the deletor and frees the object back into the pool, right? Should we name this something other than Release(), like Delete(). Because Release() that is part of ReferenceCounted does a ref counting.

Reply:
Yeah the name here is pretty bad, but it's what's used by `Deletor` in `ReferenceCounted` - poor choice of names for sure, but orthogonal to any changes here (the fact that `ReferenceCounted` uses `Release` to decrement + `Deletor::Release` to free is, I agree, terrible naming)